app/src/main/java/pl/droidevs/books/model/BookId.java

```diff
@@ -0,0 +1,13 @@
+package pl.droidevs.books.model;
+
+import android.support.annotation.NonNull;
+
+import java.io.Serializable;
+
+public class BookId implements Serializable {
```

Comment:
A tutaj bym prosiła o wyjaśnienie dlaczego w ten sposób id a nie int po prostu? :)

Reply:
Im mniej gołych typów prostych lata po aplikacji tym lepiej. W DDD bardzo często spotyka się właśnie takie podejście. Mamy hermetyzację liczby i typów pól które identyfikują nasz obiekt. Jeśli zmieni się PK obiektu w bazie danych to w ilu miejscach ta zmiana będzie widoczna? W jednym, bowiem cała aplikacja kożysta z bookId. Inna zaleta to utrudnione przekazanie niewłaściwego parametru to metod pobierających dane po idku. 
Chociaż zapomniałem o hashCode i equals :smile: I może lepsza byłaby statyczna metoda fabrykująca zamiast konstruktora: np. valueOf()

Reply:
Ooo :) A możesz polecić na ten temat jakieś źródła wiedzy i generalnie o DDD. Znam tylko generalną koncepcję. :)

Reply:
https://bottega.com.pl/materialy.xhtm?cat=DDD oraz na helionie masz dwie książki o tej tematyce, co prawda nie czytałem ich jeszcze, ale mam kupione 😃